client/components/auth-form.jsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+
+export default class AuthForm extends React.Component {
+  constructor(props) {
+    super(props);
```

Comment:
I recommend breaking this into two components. There's just about too much conditional logic. The dealbreaker for me was conditionally rendering more inputs.